hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,724 @@
+            "contact_id": new_contact.id,
+        }
+
+    @route(
+        '/update_contact/<model("res.partner"):partner>',
+        methods=["POST"],
+        type="json",
+        auth="user",
+    )
+    def update_contact(self, partner=False):
+        """Updates a contact with new values.
+
+        This function updates the details of a specified contact using the values
+        provided in the JSON request body.
+
+        JSON request body:
+            - partner_id (int): The ID of the contact to be updated.
+            - update_vals (dict): A dictionary containing the values to be updated.
+
+        JSON response:
+            - message (str): A message indicating that the contact has been successfully updated.
+
+        Returns:
+            dict: A dictionary with a success message.
+
+        """
+        update_vals = request.jsonrequest.get("update_vals")  # Values to update
+        partner.write(update_vals)  # Update the contact with the new values
+        logger.debug("Contact updated with ID %s", partner.id)
+
+        return {"message": f"Contact with ID: {partner.id} successfully updated."}
+
+    @route(
+        ["/create_delivery_address", "/create_invoice_address"],
+        methods=["POST"],
+        type="json",
+        auth="user",
+    )
+    def create_address(self):
+        """Creates or updates the billing or delivery address for a given partner.
+
+        This function creates or updates the billing or delivery address of a specified partner
+        using the details provided in the JSON request body.
+
+        JSON request body:
+            - partner_id (int): The ID of the partner.
+            - address_data (dict): A dictionary containing the address details.
+            - address_type (str): The type of address to update or create ("invoice" or "delivery").
+            - only_create (bool, optional): A boolean indicating if only creation should be done,
+            omitting the update of an existing address (default is False).
+
+        JSON response:
+            - message (str): A message indicating that the address has been successfully created or updated.
+            - address_id (int): The ID of the created or updated address.
+
+        Returns:
+            dict: A dictionary with a success message and the address ID.
+        """
+        env = request.env
+        data = request.jsonrequest
+        partner_id = data.get("partner_id")
+        address_data = data.get("address_data")
+        address_type = data.get("address_type")  # "invoice" or "delivery"
+        only_create = data.get("only_create", False)  # Boolean to omit address update
+
+        # Search for existing address of the given type
+        address = env["res.partner"].search(
+            [("parent_id", "=", partner_id), ("type", "=", address_type)], limit=1
+        )
+
+        if only_create:
+            address = False
+        if address:
```

Comment:
```suggestion
        if address and not only_create:
```

this way we can avoid the if above